controllers/replica_set_controller.go

```diff
@@ -713,7 +713,32 @@ exec mongod -f /data/automation-mongod.conf ;
 	)
 }
 
+type statefulSetSpec interface {
+	ServiceName() string
+	GetName() string
+	GetNamespace() string
+	GetVersion() string
+	AutomationConfigSecretName() string
+	GetAgentKeyfileSecretNamespacedName() types.NamespacedName
+}
+
 func buildStatefulSetModificationFunction(mdb mdbv1.MongoDBCommunity) statefulset.Modification {
+	commonModification := BuildCommonStatefulSetModificationFunction(mdb, mdb)
+	return statefulset.Apply(
+		commonModification,
+		statefulset.WithOwnerReference([]metav1.OwnerReference{getOwnerReference(mdb)}),
+		statefulset.WithUpdateStrategyType(getUpdateStrategyType(mdb)),
+		statefulset.WithPodSpecTemplate(
+			podtemplatespec.Apply(
+				buildTLSPodSpecModification(mdb),
+			),
+		),
+
+		statefulset.WithCustomSpecs(mdb.Spec.StatefulSetConfiguration.SpecWrapper.Spec),
+	)
+}
+
+func BuildCommonStatefulSetModificationFunction(mdb statefulSetSpec, scaler scale.ReplicaSetScaler) statefulset.Modification {
```

Comment:
common between what? I think this needs a docstring. Also, I don't think we should use the word common!

I think any of the following are more appropriate

`StandardReplicaSetStatefulset`, `UnconfiguredReplicaSetStatefulSet`, `ReplicaSetStatefulSet`, `MongoDBReplicaSetStatefulSet`

with a docstring or note explicitly stating which configuration is absent. What do you think?

Reply:
Also, I think in this case it makes sense to just return an `appv1.StatefulSet`, I don't really think we get much benefit from returning the function itself, the caller can make any changes required. It should result in a simpler api.

Reply:
Changed the name, but as agreed on zoom, not changing the return type for now!